client/blocks/upgrade-nudge-expanded/index.jsx

```diff
@@ -0,0 +1,145 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes, Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import classNames from 'classnames';
+import page from 'page';
+import { bindActionCreators } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import QueryPlans from 'components/data/query-plans';
+import Gridicon from 'components/gridicon';
+import PlanCompareCard from 'my-sites/plan-compare-card';
+import PlanCompareCardItem from 'my-sites/plan-compare-card/item';
+import TrackComponentView from 'lib/analytics/track-component-view';
+import formatCurrency from 'lib/format-currency';
+import { preventWidows } from 'lib/formatting';
+import { getFeatureTitle } from 'lib/plans';
+import { getPlanBySlug } from 'state/plans/selectors';
+import { PLAN_PERSONAL, getPlanClass, plansList } from 'lib/plans/constants';
+import { getCurrentPlan } from 'state/sites/plans/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteSlug } from 'state/sites/selectors';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+class ExpandedUpgradeNudge extends Component {
+	constructor( props ) {
+		super( props );
+		this.upgrade = this.upgrade.bind( this );
+		this.eventProperties = {
+			cta_size: 'expanded',
+			cta_name: props.event,
+			cta_feature: props.highlightedFeature
+		};
+	}
+
+	upgrade() {
+		this.props.recordTracksEvent( 'calypso_upgrade_nudge_cta_click', this.eventProperties );
+		if ( this.props.upgrade ) {
+			this.props.upgrade();
+		} else if ( this.props.planConstants.getPathSlug && this.props.siteSlug ) {
+			page( `/checkout/${ this.props.siteSlug }/${ this.props.planConstants.getPathSlug() }` );
+		}
+	}
+
+	render() {
+		//Display only if upgrade path available
+		if (
+			! this.props.currentPlan ||
+			( this.props.planConstants.availableFor && ! this.props.planConstants.availableFor( this.props.currentPlan.productSlug ) )
+		) {
+			return null;
+		}
+
+		const price = formatCurrency( this.props.plan.raw_price / 12, this.props.plan.currency_code );
+		const features = this.props.planConstants.getPromotedFeatures().filter(
+			feature => feature !== this.props.highlightedFeature
+		).slice( 0, 6 );
+
+		return (
+			<Card className="upgrade-nudge-expanded">
+				<QueryPlans />
+				<TrackComponentView { ...( { eventName: this.props.eventName, eventProperties: this.eventProperties } ) } />
+				<div className="upgrade-nudge-expanded__plan-card">
+					<PlanCompareCard
+						title={ this.props.plan.product_name_short }
+						line={ this.props.translate( '%(price)s per month, billed yearly', { args: { price } } ) }
```

Comment:
I can see that `this.props.translate` is used multiple times. Therefore, I think it would be better to extract it as a `const { translate } = this.props` to make the code a bit nicer. :)